.github/workflows/tests.yml

```diff
@@ -0,0 +1,87 @@
+name: Run CI tests
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  shellcheck:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run shellcheck
+        run: |
+          sudo apt-get install shellcheck
+          ./shellcheck.sh
+  
+  # This workflow contains a single job called "build"
+  test-on-linux:
+    strategy:
+      matrix:
+        os: ['ubuntu-20.04', 'ubuntu-18.04', 'ubuntu-16.04']
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run test on OS ${{ matrix.os }}
+        shell: 'script -q -e -c "bash {0}"' # work around tty issues
+        run: |
+          bash --version
+          bash -c "time bin/bats --formatter tap test"
+          # allow for some parallelity without GNU parallel, e.g. on MacOS
```

Comment:
This comment seems not to be relevant to non-MacOS, but I defer to you @martin-schulze-vireso :pray:

Reply:
A adapted the text a little